### PR TITLE
Remove reflection

### PR DESCRIPTION
**Pull request by apmasell**

### CODE REVIEW

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/AliasComparator.java [hunk 1]
```diff
@@ -36,59 +36,10 @@
...
-  public AliasComparator(Class<T> c) throws NoSuchMethodException, IllegalArgumentException {
-    method = c.getMethod("getAlias");
-    Class<?> returnClass = method.getReturnType();
-    if (returnClass.getName().equals("void")) {
-      String message = method.getName() + " has a void return type";
-      throw new IllegalArgumentException(message);
-    }
-  }
-
+public class AliasComparator<T extends Aliasable> extends AlphanumericComparator<T> {
```
Comment (djcooke) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/AliasComparator.java]: :100:

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/DateComparator.java [hunk 2]
```diff
@@ -37,87 +37,24 @@
...
-    Class returnClass = method.getReturnType();
-    if (returnClass.getName().equals("void")) {
-      String message = method.getName() + " has a void return type";
-      throw new IllegalArgumentException(message);
-    } else if (!returnClass.equals(Date.class)) {
-      String message = method.getName() + " does not return a java.util.Date";
-      throw new IllegalArgumentException(message);
-    }
-  }
+public class DateComparator<T extends QC> implements Comparator<T> {
```
Comment (djcooke) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/DateComparator.java]: Maybe rename to `QcDateComparator` for clarity

Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/DateComparator.java]: Fixed.

File: core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/Aliasable.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package uk.ac.bbsrc.tgac.miso.core.data;
+
+public interface Aliasable {
+  public String getAlias();
```
Comment (frankie13) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/Aliasable.java]: No setter? 🐕

Comment (apmasell) [inline core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/Aliasable.java]: All the implementors of the interface have a setter, but all the users of the interface are read-only. So, I don't see the need.

**Pull request merged**